tests/unit/transports/socket_transport_test.py

```diff
@@ -25,53 +25,197 @@ def test_init(self):
         assert_equals(bytearray(), self.transport._buffer)
         assert_true(self.transport._synchronous)
 
-    def test_connect_with_no_klass_arg_ipv4(self):
-        sock = mock()
-        expect(socket.socket).returns(sock)
-
+    def _set_up_connect_test(self, sock):
+        """Set up common options and expects for connect() related tests."""
         self.connection._connect_timeout = 4.12
         self.connection._sock_opts = {
             ('family', 'tcp'): 34,
             ('range', 'ipv6'): 'hex'
         }
+        expect(sock.settimeout).args(4.12).at_least_once()
+        expect(sock.setsockopt).any_order().args(
+            'family', 'tcp', 34).any_order().at_least_once()
+        expect(sock.setsockopt).any_order().args(
+            'range', 'ipv6', 'hex').any_order().at_least_once()
+        expect(sock.settimeout).args(None).at_least_once()
 
-        expect(sock.setblocking).args(True)
-        expect(sock.settimeout).args(4.12)
+    def _set_up_connect_test_fail(self, sock):
```

Comment:
`_set_up_connect_test_fail` appears to be identical to `_set_up_connect_test`. If they really are identical, let's get rid of `_set_up_connect_test_fail`